pkg/ingester/global_limits.go

```diff
@@ -0,0 +1,60 @@
+package ingester
+
+import "fmt"
+
+// GlobalLimits describes global limits used by ingester. Reaching any of these will result in Push method to return
+// (internal) error.
+type GlobalLimits struct {
+	MaxIngestionRate        float64 `yaml:"max_ingestion_rate"`
+	MaxInMemoryUsers        int64   `yaml:"max_users"`
+	MaxInMemorySeries       int64   `yaml:"max_series"`
+	MaxInflightPushRequests int64   `yaml:"max_inflight_push_requests"`
+}
+
+// Sets default limit values for unmarshalling.
+var defaultGlobalLimits *GlobalLimits = nil
```

Comment:
I find this `defaultGlobalLimits` a bit tricky to understand. Why do we need the logic using it?

Reply:
We use it to enforce use of our default values for fields, that are missing in YAML.

For example:
```yaml
ingester_limits:
  max_ingestion_rate: 100000
```

Will set `max_ingestion_rate`, but other fields will have default values -- and we want to use default values provided on command line.

Alternative to using `defaultGlobalLimits` hack would be to define fields in `GlobalLimits` struct as pointers. In that case, YAML unmarshalling of above snippet would result in:
```
GlobalLimits {
    MaxIngestionRate: pointer-to(100000),
    MaxInMemoryUsers: nil,
    MaxInMemorySeries: nil,
    MaxInflightPushRequests: nil,
}
```

This would work, but would require `nil` check and use of default value if field is `nil` everywhere where these fields are used.

I chose `defaultGlobalLimits` hack, because it is exactly the same as we already use for Limit overrides. (See `SetDefaultLimitsForYAMLUnmarshalling` function).